pkgs/development/python-modules/Theano/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, runCommandCC
+, fetchFromGitHub
+, buildPythonPackage
+, pythonOlder
+, numpy
+, six
+, scipy
+, nose
+, nose-parameterized
+, pydot_ng
+, sphinx
+, pygments
+, future
+, cudaSupport ? false
+, cudatoolkit ? null
+, pycuda ? null
+, libgpuarray ? null
+, cudnn ? null
+}:
+
+assert cudaSupport
+  -> cudatoolkit != null
+  && libgpuarray != null
+  && pycuda != null
+  && cudnn != null;
+
+buildPythonPackage rec {
+  name = "Theano-cuda-${version}";
+  version = "0.8.2";
+
+  src = fetchFromGitHub {
+    owner = "Theano";
+    repo = "Theano";
+    rev = "46fbfeb628220b5e42bf8277a5955c52d153e874";
+    sha256 = "1sl91gli3jaw5gpjqqab4fiq4x6282spqciaid1s65pjsf3k55sc";
+  };
+
+  doCheck = false;
+
+  patches = [ ./compiler.patch ];
+  postPatch =
+    let
+      # Closure for running a command in a stdenv for building with given buildInputs
+      wrapped = command: buildInputs:
```

Comment:
Perhaps this sort of function could be added the standard libraries?

I say that as I think there are actually quite a few places it could be used.  For example, I recently packaged up [COFFEE](https://github.com/coneoproject/COFFEE) which has similar issues (another python package that compiles code on the fly and then dynamically loads it into itself).

I also noticed in another package that behind the scenes nvcc was being called and it was using my system gcc.  This was failing as nvcc wasn't compatible with the system gcc version.  Ideally a Nix provided nvcc should be using a Nix provided gcc determined at the installation time of the nvcc.